src/Apps/Order/__tests__/getRedirect.test.ts

```diff
@@ -0,0 +1,115 @@
+  }
+  function aRuleWithChildren(
+    children: Array<RedirectRecord<{}>>
+  ): RedirectRecord<{}> {
+    return {
+      path: "",
+      rules: [aNonMatchingPredicate],
+      children: [...children],
+    }
+  }
+
+  it("returns null when the root rule doesn`t match", () => {
+    const rule: RedirectRecord<{}> = {
+      path: "",
+      rules: [aNonMatchingPredicate],
+    }
+    const result = getRedirect(rule, "some-path", {})
+
+    expect(result).toEqual(null)
+  })
+
+  it("returns the root path when the root rule matches", () => {
+    const rule: RedirectRecord<{}> = {
+      path: "",
+      rules: [aMatchingPredicateWithResult("the-root")],
+    }
+    const result = getRedirect(rule, "some-path", {})
+
+    expect(result).toEqual("the-root")
+  })
+
+  describe("matching child rules", () => {
+    it("doesn't match child routes when the user isn't in this section of the app", () => {
+      const rule = aRuleWithChildren([
+        {
+          path: "some-other-section",
+          rules: [aMatchingPredicateWithResult("some-other-section-route")],
+        },
+      ])
+
+      const result = getRedirect(rule, "this-section", {})
+
+      expect(result).toEqual(null)
+    })
+
+    it("matches child routes when the user is in this section of the app", () => {
+      const rule = aRuleWithChildren([
+        {
+          path: "this-section",
+          rules: [aMatchingPredicateWithResult("this-section-route")],
+        },
+      ])
+
+      const result = getRedirect(rule, "this-section", {})
+
+      expect(result).toEqual("this-section-route")
+    })
+
+    it("matches child routes when the user is deep in this section", () => {
+      const rule = aRuleWithChildren([
+        {
+          path: "this-section",
+          rules: [aMatchingPredicateWithResult("this-section-route")],
+        },
+      ])
+
+      const result = getRedirect(rule, "this-section/and/this/page", {})
+
+      expect(result).toEqual("this-section-route")
+    })
+
+    it("chooses the most specific matching child (i.e. longest path match", () => {
+      const rule = aRuleWithChildren([
+        {
+          path: "this-section",
+          rules: [aMatchingPredicateWithResult("losing-route")],
+        },
+        {
+          path: "this-section/and/this",
+          rules: [aMatchingPredicateWithResult("winning-route")],
+        },
+      ])
+
+      const result = getRedirect(rule, "this-section/and/this/page", {})
+
+      expect(result).toEqual("winning-route")
+    })
+
+    it("matches child routes despite slashes in current location", () => {
+      const rule = aRuleWithChildren([
+        {
+          path: "this-section/this-subsection",
+          rules: [aMatchingPredicateWithResult("this-section-route")],
+        },
+      ])
+
+      // REVIEW: this test fails if we change the path to
+      //  "/this-section////this-subsection/this-page"
+      // Is the intention to ignore extra slashes at ALL levels?
```

Comment:
Just checked, `found` does not normalize paths, so we probably shouldn't either. I can't imagine when it'd be necessary anyway.